### PR TITLE
Flying fish notification at minnows

### LINKED ISSUE #13040: Add "Flying Fish Notification" to Fishing plugin
I like to AFK, as such i have set my idle notification to very low while fishing minnows, and doing other stuff while fishing them, due to this i often keep fishing when the flying fish appears causing an unnessecary amount of loss due to those flying fishies.

Simply adding an option to send a notification when a flying fish appears will reduce this issue immensly, this shouldnt take much effort since there is already an option that changes the color of the fishing spot when the flying fish appears.

fishing

Thank you

### PR DESCRIPTION
**Pull request by anjaneya2b**
image

Fixes #13040

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java [hunk 1]
```diff
@@ -150,6 +150,14 @@ default boolean showMinnowOverlay()
 
 	@ConfigItem(
 		position = 10,
+		keyName = "flyingFish",
+		name = "Flying fish notification",
+		description = "Send a notification when there is a flying fish."
+	)
+	default boolean flyingFishNotification() { return true; }
```
Comment (Broooklyn) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java]:
```suggestion
	default boolean flyingFishNotification()
	{
	 	return true;
	}
```

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java [hunk 2]
```diff
@@ -150,6 +150,17 @@ default boolean showMinnowOverlay()
 
 	@ConfigItem(
 		position = 10,
+		keyName = "flyingFish",
+		name = "Flying fish notification",
+		description = "Send a notification when there is a flying fish."
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java:155]:
```suggestion
		description = "Send a notification when a flying fish spawns on your fishing spot."
```

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java [hunk 3]
```diff
@@ -215,6 +215,11 @@ public void onChatMessage(ChatMessage event)
 			spotOverlay.setHidden(false);
 			fishingSpotMinimapOverlay.setHidden(false);
 		}
+
+		if (event.getMessage().contains("A flying fish jumps up and eats some of your minnows!") && config.flyingFishNotification())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java:219]:
```suggestion
		if (event.getMessage().equals("A flying fish jumps up and eats some of your minnows!") && config.flyingFishNotification())
```

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java [hunk 4]
```diff
@@ -215,6 +215,11 @@ public void onChatMessage(ChatMessage event)
...
+		{
+			notifier.notify("A flying fish jumps up and eats some of your minnows!");
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java:221]:
```suggestion
			notifier.notify("A flying fish is eating your minnows!");
```

**Review by Nightfirecat — COMMENTED** [review]
Generally LGTM

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java [hunk 5]
```diff
@@ -150,6 +150,17 @@ default boolean showMinnowOverlay()
 
 	@ConfigItem(
 		position = 10,
+		keyName = "flyingFish",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java:153]: I'll commit this via GitHub's UI
```suggestion
		keyName = "flyingFishNotification",
```

**Pull request merged**